test/as-json.js

```diff
@@ -78,6 +78,50 @@ allocCluster.test('getting an ok response', {
     });
 });
 
+allocCluster.test('request supports baseAppHeaders', {
+    numPeers: 2
+}, function t(cluster, assert) {
+    var tchannelJSON = makeTChannelJSONServer(cluster, {
+        okResponse: true
+    });
+
+    var server = cluster.channels[0].subChannels.server;
+    tchannelJSON.register(server, 'test', {}, testResponse);
+
+    var client = cluster.channels[1].subChannels.server;
+
+    tchannelJSON.request({
+        serviceName: 'server',
+        hasNoParent: true,
+        baseAppHeaders: {key: 'val'},
```

Comment:
Let’s worry about this name some more. `default` seems more idiomatic than `base` (though I’m sure you considered default first), and i would like to avoid introducing the `App` abbreviation. Let’s classify headers as “transport headers”, “request headers”, and “context headers”. `defaultRequestHeaders` seems like a step better.